pkgs/os-specific/linux/musl/default.nix

```diff
@@ -1,11 +1,33 @@
-{ stdenv, fetchurl, fetchpatch }:
+{ stdenv, lib, fetchurl
+, buildPlatform
+, hostPlatform
+, buildPackages
+, linuxHeaders ? null
+, useBSDCompatHeaders ? true
+}:
+let
+  cdefs_h = fetchurl {
+    url = "http://git.alpinelinux.org/cgit/aports/plain/main/libc-dev/sys-cdefs.h";
+    sha256 = "16l3dqnfq0f20rzbkhc38v74nqcsh9n3f343bpczqq8b1rz6vfrh";
+  };
+  queue_h = fetchurl {
+    url = "http://git.alpinelinux.org/cgit/aports/plain/main/libc-dev/sys-queue.h";
+    sha256 = "12qm82id7zys92a1qh2l1qf2wqgq6jr4qlbjmqyfffz3s3nhfd61";
+  };
+  tree_h = fetchurl {
+    url = "http://git.alpinelinux.org/cgit/aports/plain/main/libc-dev/sys-tree.h";
+    sha256 = "14igk6k00bnpfw660qhswagyhvr0gfqg4q55dxvaaq7ikfkrir71";
+  };
 
+  cross = if buildPlatform != hostPlatform then hostPlatform else null;
+in
 stdenv.mkDerivation rec {
-  name    = "musl-${version}";
+  name    = "musl-${version}" +
+    lib.optionalString (cross != null) "-${cross.config}";
```

Comment:
Adapter should already do this, right? (I recently removed redundency from glibc.)

Reply:
Oh! Yes adapter of course does do this, I assumed/guessed this was for libcCross which is a bit "special".

That said, it does result in a lot of redundancy as-is because "build != host" is not how you check "am I being built as libcCross"!  Also it might not be needed even then...yep it sure doesn't appear to be needed then either.

Ah yes, I probably was following along glibc, from before that change.

I'll remove this, although immediately reverting it to avoid mass-rebuild while working on other issues.